packages/@eventual/core-runtime/src/local/stores/bucket-store.ts

```diff
@@ -0,0 +1,220 @@
+    Record<
+      string,
+      {
+        body: Uint8Array | string | Buffer;
+        contentLength: number;
+        etag: string;
+      }
+    >
+  > = {};
+
+  // TODO: publish events for the stream
+  constructor(private props: LocalBucketStoreProps) {}
+  async get(
+    bucketName: string,
+    key: string,
+    options?: GetBucketObjectOptions
+  ): Promise<GetBucketObjectResponse | undefined> {
+    const bucket = this.objects[bucketName];
+
+    if (!bucket) {
+      return undefined;
+    }
+
+    const object = bucket[key];
+
+    if (!object) {
+      return object;
+    }
+
+    if (options?.etag && options?.etag !== object.etag) {
+      return undefined;
+    }
+
+    const stream = new Stream.Readable();
+    stream.push(object.body);
+
+    return {
+      ...object,
+      body: stream,
+    };
+  }
+
+  async put(
+    bucketName: string,
+    key: string,
+    data: string | Buffer | Readable
+  ): Promise<PutBucketObjectResponse> {
+    const bucket = (this.objects[bucketName] ??= {});
+
+    const body =
+      typeof data === "string"
+        ? data
+        : data instanceof Buffer
+        ? data
+        : (data.read() as Buffer);
+
+    const etag = getEtag(body);
+
+    bucket[key] = {
+      body,
+      contentLength: body.length,
+      etag,
+    };
+
+    this.props.localConnector.pushWorkflowTask({
+      bucketName,
+      key,
+      operation: "put",
+      size: body.length,
+      etag,
+    });
+
+    return {
+      etag,
+    };
+  }
+
+  async delete(bucketName: string, key: string): Promise<void> {
+    const bucket = this.objects[bucketName];
+
+    if (!bucket) {
+      return;
+    }
+
+    delete bucket[key];
+
+    this.props.localConnector.pushWorkflowTask({
+      bucketName,
+      key,
+      operation: "delete",
+    });
+  }
+
+  async copyTo(
+    bucketName: string,
+    key: string,
+    sourceKey: string,
+    sourceBucket?: Bucket | undefined,
+    options?: CopyBucketObjectOptions | undefined
+  ): Promise<CopyBucketObjectResponse> {
+    const _sourceBucket = this.objects[sourceBucket?.name ?? bucketName];
+
+    if (!_sourceBucket) {
+      return {};
+    }
+
+    const sourceObject = _sourceBucket[sourceKey];
+
+    if (
+      !sourceObject ||
+      (options?.sourceEtag && options.sourceEtag !== sourceObject.etag)
+    ) {
+      return {};
+    }
+
+    const destBucket = (this.objects[bucketName] ??= {});
+
+    destBucket[key] = sourceObject;
+
+    this.props.localConnector.pushWorkflowTask({
+      bucketName,
+      key,
+      operation: "copy",
+      size: sourceObject.contentLength,
+      etag: sourceObject.etag,
+    });
+
+    return {
+      etag: sourceObject.etag,
+    };
+  }
+
+  presignedUrl(
+    _bucketName: string,
+    _key: string,
+    _operation: "get" | "put" | "delete",
+    _expires?: DurationSchedule | undefined
+  ): Promise<{ url: string; expires: string }> {
+    throw new Error("Presigned urls are not supported in Eventual Local");
```

Comment:
What would it take to support this?

Reply:
I think we'd need a system path that supports the president contract.

Reply:
Let's track?

Reply:
https://github.com/functionless/eventual/issues/341